Кабинет менеджера, новости

dshishkaryan opened the pull request:
Дать возможность редактировать дату публикации, если новость еще не опубликована!
На странице новости: не выводить кнопки Вход и Регистрация, если пользователь вошел в систему

Добавить поле - дата изменения, изменять при каждом редактировании новости.

dshishkaryan committed:
added List<News> findByStatusEquals(boolean status);

dshishkaryan committed:
added new mappings

dshishkaryan committed:
added new exception class

dshishkaryan committed:
added script

dshishkaryan committed:
added status ro newsinit

dshishkaryan committed:
moved bootstrap scripts down

dshishkaryan committed:
refactor according new mapping

dshishkaryan committed:
code style

dshishkaryan committed:
added private boolean status; private LocalDateTime modifiedDate;

dshishkaryan committed:
renamed controller

dshishkaryan committed:
modified div for news

dshishkaryan committed:
added
- function which render div with news
- function which defines which button was clicked
- function which handles edit button

dshishkaryan committed:
added public ResponseEntity<News> getNews(@PathVariable Long id) method
changed delete mapping method

dshishkaryan committed:
changed find by id method to return News

dshishkaryan committed:
bugfix (don't check close or submit button)
modified java docs
code style

dshishkaryan committed:
small changes in data init

dshishkaryan committed:
bug fixed

dshishkaryan committed:
changed field

dshishkaryan committed:
small changes

dshishkaryan committed:
deleted new modal

dshishkaryan committed:
added a lot of functions

dshishkaryan committed:
deleted List<News> findByStatusEquals(boolean status);

dshishkaryan committed:
changed mappings

dshishkaryan committed:
before changing bootstrap scripts

dshishkaryan committed:
code style

dshishkaryan committed:
changed LocalDateTime to LocalDate

dshishkaryan committed:
removed unnecessary if

dshishkaryan committed:
deleted unnecessary scripts

dshishkaryan committed:
added POST fetch

dshishkaryan committed:
code style

dshishkaryan committed:
Merge branch 'dev' into feature/manager_cabinet_news

dshishkaryan committed:
промежуточный коммит

dshishkaryan committed:
changed methods to throw exceptions if there are no value
added List<News> getAllArchivedNews();
changed methods in repository to check is new archived or not

dshishkaryan committed:
added archived checkbox

dshishkaryan committed:
changes with date input field

dshishkaryan committed:
added public ResponseEntity<List<News>> getAllArchivedNews() and code style

dshishkaryan committed:
some changes

dshishkaryan committed:
done

dshishkaryan committed:
Merge branch 'dev' into feature/manager_cabinet_news

tediik commented on src/main/resources/static/js/newsManagementRenderAndFunctions.js:
```diff
@@ -0,0 +1,385 @@
+    fetchNews("/all", '#allNews')
+});
+$('#editNewsModal').on('hidden.bs.modal', function () {
+    fetchNews("/all", '#allNews')
+    fetchNews("/published", '#publishedNews')
+    fetchNews("/unpublished", '#unpublished')
+    fetchNews("/archived", '#archived')
```
проблема в том, что четыре отдельных запроса будут выполнятся дольше, чем 1, в котором мы получим все типы новостей сразу

dshishkaryan replied:
Конкретно в этом случае я не придумал как лучше можно данный функционал реализовать. Задача в том что если мы например ставим галочку архивировать на любой новости из любого раздела, что бы при закрытии модального окна она убиралась из текущего раздела и отрисовывалась в новом. В теории наверное можно заморочиться, определять из какого она раздела, удалять в нем и рисовать в разделе архивные.

dshishkaryan replied:
Наверняка есть какой то оптимальный способ, но в голову что от не приходит. Можно сделать вот так location.reload(), что тоже наверное не лучший вариант.

tediik commented on src/main/resources/static/js/newsManagementRenderAndFunctions.js:
```diff
@@ -0,0 +1,385 @@
+ * function clears modal window fields
+ */
+function clearModalFields() {
+    document.getElementById('idNewsUpdate').value = ''
+    document.getElementById('titleNewsUpdate').value = ''
+    document.getElementById('anonsNewsUpdate').value = ''
+    $('#fullTextUpdate').summernote('code', '')
+    document.getElementById('postingDateUpdate').value = ''
+    document.getElementById('archiveCheckbox').checked = false
```
а если сделать все эти поля элементами формы <form> то потом можно вызвать буквально один метод reset и сбросить форму целиком

dshishkaryan replied:
скорее всего ты прав, но не уверен что это сработает когда в форме внешняя библиотека summrnote.
Там особым образом очищается текст $('#fullTextUpdate').summernote('code', '').  Делать форму для 5 полей и отдельно очищать одно?
Но если это критичный момент, конечно переделаю на форму.

tediik replied:
Ну, если бы это было не важно, я бы не стал писать замечание, переделай на форму, а одно неподдающееся общей логике поле - очищай вручную

dshishkaryan replied:
ready

dshishkaryan replied:
да, так действительно удобней очищать поля. да и редактор тоже очистился нормально.

tediik commented on src/main/java/com/jm/online_store/service/impl/NewsServiceImpl.java:
```diff
@@ -42,15 +46,15 @@ public void save(News news) {
-    public Optional<News> findById(long id) {
-        return newsRepository.findById(id);
+    public News findById(long id) {
+        News news = newsRepository.findById(id).orElseThrow(NewsNotFoundException::new);
+        return news;
```
сделай в 1 строчку

dshishkaryan replied:
исправил

dshishkaryan committed:
removed :
    fetchNews("/all", '#allNews')
    fetchNews("/published", '#publishedNews')
    fetchNews("/unpublished", '#unpublished')
    fetchNews("/archived", '#archived')

dshishkaryan committed:
code style

dshishkaryan committed:
modified clearModalFields() function. Replaced with $('form[name=editNewsFormModal]').trigger('reset')
